exchanges/graphcache/src/types.ts

```diff
@@ -128,6 +128,13 @@ export interface Cache {
     data: T,
     variables?: V
   ): void;
+
+  link(
+    entity: Data | string,
+    field: string,
+    entities: Data | Data[],
```

Comment:
I think this current signature is a little confusing. Originally I proposed this because it's more intuitive:

```ts
link(
  entity: string | Data | null,
  fieldName: string,
  fieldArgs: Variables | null,
  link: Link<Data | string>
): void;

link(
  entity: string | Data | null,
  fieldName: string,
  link: Link<Data | string>
): void;
```

The overload helps here in case people use `cache.keyOfField` or have no arguments. It also helps that arguments come after the field name.

The `Link` type can be reused here for nested lists and for allowing `string | Data | null`.

Reply:
We also have to check here what happens when scalars are passed in accidentally or what the common case is for partial data, since it becomes more likely for people to chain `cache.link` and write `null` data.

I'm not sure whether in that case we need to make special arrangements to write `undefined` instead of `null`